Approved/3-multidimensional-arrays.md

```diff
@@ -0,0 +1,701 @@
+Mixing 1D and multidimensional arrays.
+
+```qsharp
+// Here, we declare the first two levels of indexing as being rectangular,
+// resulting in a three-dimensional array.
+let data = [
+    [|
+        // After rows, [] denote arrays as elements again.
+        [| [0], [1, 2] |],
+        [| [3, 4, 5], [6, 7, 8, 9] |]
+    |],
+
+    [|
+        [| [10, 11, 12, 13, 14], [15, 16, 17, 18, 19, 20] |],
+        [| [21, 22, 23, 24, 25, 26, 27], [28, 29, 30, 31, 32, 33, 34, 35] |]
+    |]
+];
+// data: [||[Int]||] (that is, a three-dimensional array of arrays of integers)
+// Get the zero-th "plane," first "row", zeroth "column," and third element.
+Message($"{data[0, 1, 0][2]}"); // ← 5
+```
+
+Example 3:
+Using expressions as subarrays of multidimensional arrays results in a compile-time error.
+
+```qsharp
+let a = [2, 3];
+// The following is a compile-time error, since `a` is not a 1D array literal
+// of length 2.
+let data = [
+    [| 0, 1 |],
+    a
+];
+// Using a new library function that can `fail` at runtime works, however.
+let data = Concatenated(0, // concatenate along the 0th (row) axis
+    [ [| 0, 1 |] ],
+    a
+);
+// data: [|Int|] = [
+//     [|0, 1|],
+//     [|2, 3|]
+// ];
+```
+
+Example 4:
+Using the copy-and-replace operator to manipulate multidimensional arrays.
+
+```qsharp
+function ElementaryMatrix(
+    (nRows : Int, nCols : Int), (idxRow : Int, idxCol : Int)
+) : [|Double|] {
+    return ConstantArray2((nRows, nCols), 0.0) w/ (idxRow, idxCol) <- 1.0;
+}
+```
+
+Example 5:
+Slicing multidimensional arrays by ranges.
+
+```qsharp
+let data = [
+    [|0, 1, 2|],
+    [|3, 4, 5|],
+    [|6, 7, 8|]
+];
+
+// Slicing an index by a Range does not reduce the dimensionality
+// of the resulting array.
+let corners = data[0..2..2, 0..2..2];
+// corners: [|Int|] = [[|0, 2|], [|6, 8|]]
+
+// Subscripting by an Int reduces the dimensionality of the resulting array;
+// here, since our index has one Int, the dimensionality reduces from
+// [|Int|] to [Int].
+let firstRow = data[0, ...];
+// firstRow: [Int] = [0, 1, 2]
+// The same pattern holds no matter which index we subscript with an Int.
+let firstColumn = data[..., 0];
+// firstColumn = [0, 3, 6]
+
+let data3 = [
+    [|
+        [|0, 1, 2|],
+        [|3, 4, 5|],
+        [|6, 7, 8|]
+    |],
+
+    [|
+        [|9, 10, 11|],
+        [|12, 13, 14|],
+        [|15, 16, 17|]
+    |],
+
+    [|
+        [|18, 19, 20|],
+        [|21, 22, 23|],
+        [|24, 25, 26|]
+    |]
+];
+let corners3 = data3[0..2..2, 0..2..2, 0..2..2];
+// corners3: [||Int||]
+let firstPlane = data3[0, ..., ...];
+// firstPlane: [|Int|]
+let firstRowOfFirstPlane = data3[0, 0, ...];
+// firstRowOfFirstPlane: [Int] = [0, 1, 2]
+```
+
+Example 6:
+Using multidimensional slices in copy-and-update expressions.
+
+```qsharp
+let zeros = ConstantArray2((3, 3), 0);
+
+let withCorners = zeros w/ (0..2..2, 0..2..2) <- #[[1, 2], [3, 4]];
+// withCorners: [|Int|] = [ [|1, 0, 2|], [|0, 0, 0|], [|3, 0, 4|] ]
```

Comment:
Same here regarding the type = value notation; could you please update these?

Reply:
Done.